spec/unit/ssl/state_machine_spec.rb

```diff
@@ -7,12 +7,12 @@
   include PuppetSpec::Files
 
   let(:machine) { described_class.new }
-  let(:cacert_pem) { File.read(fixtures('unit/ssl/ssl_provider/ca.pem')) }
-  let(:cacert) { OpenSSL::X509::Certificate.new(cacert_pem) }
+  let(:cacert_pem) { cacert.to_pem }
+  let(:cacert) { cert_fixture('ca.pem') }
```

Comment:
Does this work? It looks like `cacert` isn't defined until line 11

Reply:
The block for a `let(<foo>)` is lazily evaluated the first time `<foo>` is used in a test, so having these two out of order works fine. `let!(<foo>)` would force the block to be evaluated where it's defined.

Reply:
oooh, that's magic. Thanks @jhelwig !